autoscheduler/scraper/views.py

```diff
@@ -14,3 +21,61 @@ def get_object(self):
         course_num = self.request.query_params.get('course_num')
         term = self.request.query_params.get('term')
         return Course.objects.get(dept=dept, course_num=course_num, term=term)
+
+class ListSectionView(generics.ListAPIView):
+    """ API endpoint for viewing course information, used by /api/sections.
+        This view returns a serialized section, should
+        return list of all sections for a given course.
+    """
+    serializer_class = SectionSerializer
+
+    def get_queryset(self):
+        """ Overrides default behavior of get_queryset() to work without a primary key """
+        dept = self.request.query_params.get('dept')
+        course_num = self.request.query_params.get('course_num')
+        term = self.request.query_params.get('term')
+        return Section.objects.filter(subject=dept, course_num=course_num, term_code=term)
+
+class RetrieveTermView(generics.ListAPIView):
+    """ API endpoint for viewing terms, used by /api/term.
+        This view returns all the terms
+    """
+    def list(self, request):
+        """Overrides default behavior of list method so terms are ouput in
+           the format {"201831": "Fall 2018 - College Station", ...} Does this by creating
+           a new ordered dictionary called formatted_data"""
+        queryset = self.get_queryset()
+        serializer = TermSerializer(queryset, many=True)
+        formatted_data = OrderedDict()
+        for i in serializer.data:
+            formatted_data[i['term']] = i['desc']
+        return Response(formatted_data)
+
+    queryset = Department.objects.all().distinct('term').order_by('-term')
+    serializer_class = TermSerializer
+
+class RetrieveCourseSearchView(generics.ListAPIView):
+    """ API endpoint for viewing list of courses searched off of
+        searchText parameter"""
+    def get_queryset(self):
+        """ Overrides default behavior of get_queryset() to work using
+            search and term parameter in the url"""
+        search = self.request.query_params.get('search')
+        term = self.request.query_params.get('term')
+        return Course.objects.filter(id__startswith=search.replace(" ", ""), term=term)

```

Comment:
```suggestion
        return Course.objects.filter(id__startswith=search.replace("%20", ""), term=term)

```
Since we're getting the text from a URL, spaces will be encoded as `%20`

Reply:
We'll also want to convert the search to uppercase, so that for example if the user searches "csce 313", "CSCE 313" will show up